scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -1251,21 +1247,11 @@ def func(*args, **kwargs):
         res2 = func(a_masked2, weights=b_masked2, nan_policy="omit", axis=axis)
         res3 = func(a_masked3, weights=b_masked3, nan_policy="raise", axis=axis)
         res4 = func(a_masked3, weights=b_masked3, nan_policy="propagate", axis=axis)
-        # Would test with a_masked3/b_masked3, but there is a bug in np.average
-        # that causes a bug in _no_deco mean with masked weights. Would use
-        # np.ma.average, but that causes other problems. See numpy/numpy#7330.
-        if weighted_fun_name in {"hmean"}:
-            weighted_fun_ma = getattr(stats.mstats, weighted_fun_name)
-            res5 = weighted_fun_ma(a_masked4, weights=b_masked4,
-                                   axis=axis, _no_deco=True)
```

Comment:
The decorator implements the required behavior, so we've removed the native implementation. This means that calling with `_no_deco` is no longer a reference we can test against. No matter - this test was redundant anyway.